contracts/core/SetToken.sol

```diff
@@ -1,110 +1,126 @@
+import { SafeMath } from "zeppelin-solidity/contracts/math/SafeMath.sol";
 import "../lib/AddressArrayUtils.sol";
-import "./interfaces/ISetToken.sol";
 
 
 /**
  * @title {Set}
  * @author Felix Feng
+ *
  * @dev Implementation of the basic {Set} token.
+ *
  */
 contract SetToken is
     StandardToken,
-    DetailedERC20("", "", 18),
-    ISetToken
+    DetailedERC20("", "", 18)
 {
     using SafeMath for uint256;
     using AddressArrayUtils for address[];
 
-    ///////////////////////////////////////////////////////////
-    /// Data Structures
-    ///////////////////////////////////////////////////////////
+    /* ============ Constants ============ */
+    
+    string constant COMPONENTS_INPUT_MISMATCH = "Components and units must be the same length.";
+    string constant COMPONENTS_MISSING = "Components must not be empty.";
+    string constant UNITS_MISSING = "Units must not be empty.";
+    string constant ZERO_QUANTITY = "Quantity must be greater than zero.";
+
+    /* ============ Structs ============ */
+
     struct Component {
         address address_;
         uint unit_;
     }
 
-    ///////////////////////////////////////////////////////////
-    /// States
-    ///////////////////////////////////////////////////////////
+    /* ============ State Variables ============ */
+
     uint public naturalUnit;
     Component[] public components;
 
     // Mapping of componentHash to isComponent
     mapping(bytes32 => bool) internal isComponent;
-    // Mapping of index of component -> user address -> balance
-    mapping(uint => mapping(address => uint)) internal unredeemedBalances;
-
-
-    ///////////////////////////////////////////////////////////
-    /// Events
-    ///////////////////////////////////////////////////////////
-    event LogPartialRedemption(
-        address indexed _sender,
-        uint _quantity,
-        bytes32 _excludedComponents
-    );
-
-    event LogRedeemExcluded(
-        address indexed _sender,
-        bytes32 _components
-    );
-
-    ///////////////////////////////////////////////////////////
-    /// Modifiers
-    ///////////////////////////////////////////////////////////
-    modifier hasSufficientBalance(uint quantity) {
-        // Check that the sender has sufficient components
-        // Since the component length is defined ahead of time, this is not
-        // an unbounded loop
-        require(balances[msg.sender] >= quantity, "User does not have sufficient balance");
+
+    /* ============ Modifiers ============ */
+
+    modifier isMultipleOfNaturalUnit(uint _quantity) {
+        require((_quantity % naturalUnit) == 0);
         _;
     }
 
-    modifier validDestination(address _to) {
-        require(_to != address(0));
-        require(_to != address(this));
+    // Confirm that all inputs for creating a set are valid
+    modifier areValidCreationParameters(address[] _components, uint[] _units) {
```

Comment:
Perhaps we want to do these checks in the `Factory` vs. in the newly created `SetToken` contract